module-2/backend/week_two.md

```diff
@@ -5,17 +5,77 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. At a high level, what is ActiveRecord? What does it do/allow you to do?
+
+  Active Record is an object relational mapper. It maps Ruby objects to a SQL database. ActiveRecord also provides a set of methods (called on a Ruby class) to query a SQL database (translates the ruby method to SQL).
+  
 2. What kind of methods are `belongs_to`, and `has_many`? (i.e. class or instance) Give an example.
+
+  They are class methods.
+  ```ruby
+  class Student
+    has_many :classes
+    belongs_to :school
+   end
+  ```
 3. What do they allow you to do?
+
+  They allow you to create relationships between objects by providing a set of instance methods.
+  ```ruby
+  Student.first.classes
+  Student.first.school
+  ```
 4. What's the difference between agile workflow and waterfall method?
+
+  Agile workflow is doing sprints to complete discrete features. At the end of every sprint something is produced in its entirety (even if it is small). Waterfall is to complete all like parts at the same time (like all erb files) and to slowly up to the whole product.
+
 5. What is the difference between `#new` and `#create`?
+
+  `#new` makes the object but does not save it in the database. `#create` makes and saves the object.
+  
 6. At a basic level, what does cURL allow you to do?
+
+  cURL allows you to make an HTTP GET request from the command line and read the html response.
+
 7. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+
+  There is a many to many relationship between teachers and students.
+  
+  ```ruby
+  class Student
+    has_many StudentTeachers
+    has_many Teachers, through: StudentTeachers
+  end
+  
+  class StudentTeacher
+    belongs_to Student
+    belongs_to Teacher
+   end
+  
+  class Teacher
+    has_many StudentTeachers
+    has_many Students, through: StudentTeachers
+  end
+  
+  ```
+  (I drew it out in my notebook)
+  
 8. Define foreign key, primary key, and schema.
+
+  Foreign key: Key within an object that identifies a different object
+  Primary key: Key that uniquely identifies an object (self)
+  Schema: Blueprint to make the database (updated with migrations) 
+  
 9. Describe the relationship between a foreign key on one table and a primary key on another table.
+  
+  A foreign key on one table could be the same as a primary key on another table. They could both be pointing at the same object.
+
 10. What are the parts of an HTTP response?
+
+  An HTTP response has a head and a body. The head specifies the details of the message (VERB, protocol, status code, length...) and the body is directions for the browser (what to display or do).
```

Comment:
Don't believe the headers in the response need the verb (believe that's included in the request), but the first line of the headers is the status (e.g. 200, 404, etc.)